### DISCUSSION TITLE
[idea] Import and export Bloomfilter in/from kvrocks in other format

### DISCUSSION THREAD

**Discussion by adulau** [apache/kvrocks#1924] [General]
It's just an idea as we are generating different Bloomfilters from kvrocks, but we usually iterate over all the keys and generate the specific Bloomfilter.

The idea would be to import and export Bloomfilters in/from kvrocks in other formats. Importing existing Bloomfilters or even exporting them might be possible, although I suppose the data model behind kvrocks differs, especially regarding the hash functions used.

There is a commonly used format created by DCSO and their different implementations:

- [C Implementation - https://github.com/hashlookup/fleur](https://github.com/hashlookup/fleur)
- [Go Implementation - https://github.com/DCSO/bloom](https://github.com/DCSO/bloom)
- [Python Implementation - https://github.com/DCSO/flor](https://github.com/DCSO/flor)

**Comment by mapleFU** [comment]
We're using BlockSplitBloomFilter with xxhash, just same as parquet is using( without metadata ) 🤔

Besides, I think different project might uses different format for BloomFilter, and some format also have some metadata in BloomFilter in binary format

**Reply by adulau** [reply]
Indeed, so the export/import in BlockSplitBloomFilter format would be the easier? than having other format to export to.

**Reply by mapleFU** [reply]
https://github.com/apache/kvrocks-website/pull/182

Generally I think it's a bit hard to export / import, because BF is not set, we don't actually know how element is in them. I've write a doc about it's internal, you can take a look first
